### DISCUSSION TITLE
Is it possible to detect a key press event ?

### DISCUSSION THREAD

**Discussion by colinrees** [google/mesop#552] [Q&A] [answered]
Is it possible to detect a key press event ?.

I am trying to build a LLM data cleansing app and want to be able to press a button using a key press as opposed to a mouse click ?

**Comment by wwwillchen** [comment, marked as answer]
I'd recommend creating a web component: https://google.github.io/mesop/web_components/ - then you can detect a keypress event by writing JS code, e.g.

```
document.addEventListener('keydown', function(event) {
  const keyName = event.key;

  // Check if the pressed key is "a"
  if (keyName === 'a') {
    console.log("You pressed the 'a' key!"); 
  }

  // Example: Detect the Enter key
  if (keyName === 'Enter') {
    console.log("You pressed Enter!");
  }
});
```

You can [trigger an event](https://google.github.io/mesop/web_components/quickstart/#triggering-an-event) from your web component which you can use to call a Python function (e.g. the on_click event handler or an extracted function).  We've considered supporting keyboard shortcuts in a more first-class way in Mesop, but it'll take us some time to get there.
